app/controllers/CyaMaintainProfileController.scala

```diff
@@ -70,39 +76,43 @@ class CyaMaintainProfileController @Inject() (
           ).flatten
         )
         Ok(view(list, routes.CyaMaintainProfileController.onSubmitUkimsNumber))
+          .removingFromSession(dataUpdated, pageUpdated, dataRemoved, dataAdded)
       case Left(errors) =>
         logErrorsAndContinue(errorMessage, routes.ProfileController.onPageLoad(), errors)
     }
   }
 
   def onSubmitUkimsNumber(): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
-      TraderProfile.validateUkimsNumber(request.userAnswers) match {
-        case Right(value) =>
-          for {
-            traderProfile <- traderProfileConnector.getTraderProfile(request.eori)
-            _             <-
-              auditService
-                .auditMaintainProfile(traderProfile, traderProfile.copy(ukimsNumber = value), request.affinityGroup)
-            _             <- traderProfileConnector.submitTraderProfile(traderProfile.copy(ukimsNumber = value), request.eori)
-          } yield Redirect(navigator.nextPage(CyaMaintainProfilePage, NormalMode, request.userAnswers))
-        case Left(errors) =>
-          Future.successful(logErrorsAndContinue(errorMessage, routes.ProfileController.onPageLoad(), errors))
+      (for {
```

Comment:
refactored each submit so now they all follow the same pattern and are all in a for comprehension for easy reading

flow is:
validate -> get old trader profile -> build new trader profile -> audit -> submit if there is a change